sarif/include/gap/sarif/definitions.hpp

```diff
@@ -0,0 +1,2866 @@
+// Copyright 2024-present, Trail of Bits, Inc.
+
+#pragma once
+
+#include <cstdint>
+#include <forward_list>
+#include <optional>
+#include <string>
+#include <unordered_map>
+
+#include <nlohmann/json.hpp>
+
+namespace gap::sarif::definitions
+{
+    enum class version_enum {
+        k2_1_0,
+    };
+
+    NLOHMANN_JSON_SERIALIZE_ENUM(version_enum, {
+        { version_enum::k2_1_0, "2.1.0" },
+    })
+    struct run_struct;
+    struct external_properties_struct;
+
+    //
+    // Key/value pairs that provide additional information about the object.
+    //
+    struct property_bag_struct {
+        //
+        // A set of distinct strings that provide additional information.
+        //
+        std::optional<std::forward_list<std::string>> tags = std::nullopt;
+
+        std::unordered_map<std::string, nlohmann::json> additional_properties;
```

Comment:
why map to json? json is already a map

Reply:
Agreed in this case -- in general though a `std::unordered_map<std::string, std::string>` is more type-safe